ts/features/bonus/cgn/components/merchants/CgnMerchantsFilters.tsx

```diff
@@ -0,0 +1,201 @@
+            <H2>{I18n.t("bonus.cgn.merchantsList.filter.searchTitle")}</H2>
+            <Item>
+              <Input
+                value={searchValue}
+                autoFocus={true}
+                onChangeText={setSearchValue}
+                placeholderTextColor={IOColors.bluegreyLight}
+                placeholder={I18n.t(
+                  "bonus.cgn.merchantsList.filter.inputPlaceholder"
+                )}
+              />
+              <IconFont name="io-search" color={IOColors.bluegrey} />
+            </Item>
+            <View spacer large />
+            {props.isLocal && (
+              <>
+                <H2>{I18n.t("bonus.cgn.merchantsList.filter.addressTitle")}</H2>
+                <Item>
+                  <Input
+                    value={address}
+                    autoFocus={true}
+                    onChangeText={setAddress}
+                    placeholderTextColor={IOColors.bluegreyLight}
+                    placeholder={I18n.t(
+                      "bonus.cgn.merchantsList.filter.addressPlaceholder"
+                    )}
+                  />
+                  <IconFont name="io-place" color={IOColors.bluegrey} />
+                </Item>
+                <View spacer />
+                <DistanceSlider
+                  value={distance}
+                  onValueChange={setDistance}
+                  disabled={address.length === 0}
+                />
+                <View spacer large />
+              </>
+            )}
+            <H2>{I18n.t("bonus.cgn.merchantsList.filter.categories")}</H2>
+            <View spacer small />
+            <FlatList
+              data={categories}
+              keyExtractor={cat => cat.type}
+              keyboardShouldPersistTaps={"handled"}
+              ItemSeparatorComponent={() => (
+                <ItemSeparatorComponent noPadded={true} />
+              )}
+              renderItem={(listItem: ListRenderItemInfo<Category>) => (
+                <CategoryCheckbox
+                  text={I18n.t(listItem.item.nameKey)}
+                  value={listItem.item.type}
+                  checked={checkedCategories.indexOf(listItem.item.type) !== -1}
+                  onPress={onCheckBoxPress}
+                  icon={listItem.item.icon}
+                />
+              )}
+            />
+            <View spacer large />
+            {props.isLocal && (
+              <>
+                <H2>{I18n.t("bonus.cgn.merchantsList.filter.orderTitle")}</H2>
+                <View spacer small />
+                <FlatList
+                  data={orders}
+                  keyExtractor={ord => ord.value}
+                  keyboardShouldPersistTaps={"handled"}
+                  ItemSeparatorComponent={() => (
+                    <ItemSeparatorComponent noPadded={true} />
+                  )}
+                  renderItem={(listItem: ListRenderItemInfo<OrderType>) => (
+                    <OrderOption
```

Comment:
I noticed that on first tap it doesn't respond

https://user-images.githubusercontent.com/822471/118505053-59532f80-b72c-11eb-86d2-48cf3b894016.mov

Reply:
the same for the checkboxes

Reply:
Solved in ef84976